lib/chef_handler_foreman/foreman_hooks.rb

```diff
@@ -10,9 +10,16 @@
 
 module ChefHandlerForeman
   module ForemanHooks
+
+    # Provide a chef-client cookbook friendly option
+    def foreman_server_url(url)
+      foreman_server_options(:url => url)
+    end
+
     # {:url => '', ...}
-    def foreman_server_options(options)
-      options = { :client_key => client_key || '/etc/chef/client.pem' }.merge(options)
+    def foreman_server_options(options={})
+      options[:client_key] = '/etc/chef/client.pem' unless options[:client_key]
```

Comment:
This drops default value coming from chef config that we retrieve by calling `client_key`.